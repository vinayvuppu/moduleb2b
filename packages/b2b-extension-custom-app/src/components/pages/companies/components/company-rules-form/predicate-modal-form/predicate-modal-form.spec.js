import React from 'react';
import { shallow } from 'enzyme';
import PredicateModalForm from './predicate-modal-form';

jest.mock('react-intl', () => {
  const actual = jest.requireActual('react-intl');
  return {
    ...actual,
    useIntl: jest.fn(() => ({
      formatMessage: jest.fn(() => 'formatted message'),
    })),
  };
});

jest.mock('@commercetools-frontend/permissions', () => {
  const actual = jest.requireActual('@commercetools-frontend/permissions');
  return {
    ...actual,
    useIsAuthorized: jest.fn(() => true),
  };
});

jest.mock('formik', () => {
  const actual = jest.requireActual('formik');
  return {
    ...actual,
    useFormik: jest.fn(data => ({
      setFieldValue: jest.fn(),
      handleSubmit: data.onSubmit,
      handleChange: jest.fn(),
      handleBlur: jest.fn(),
      handleReset: jest.fn(),
      isSubmitting: false,
      touched: {},
      errors: {},
      values: {
        name: '',
        value: '',
      },
    })),
  };
});

const createTestProps = props => ({
  isOpen: true,
  close: jest.fn(),
  handleRemove: jest.fn(),
  handleSubmit: jest.fn(),
  initialValues: {
    name: '',
    value: '',
  },
  ...props,
});

describe('rendering', () => {
  let props;
  let wrapper;
  beforeEach(() => {
    props = createTestProps();
    wrapper = shallow(<PredicateModalForm {...props} />);
  });

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
