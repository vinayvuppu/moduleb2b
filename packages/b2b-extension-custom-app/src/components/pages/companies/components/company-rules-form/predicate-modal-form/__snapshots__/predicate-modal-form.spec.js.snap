// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering should match snapshot 1`] = `
<FormModalPage
  customControls={
    <Inline
      alignItems="flex-start"
      justifyContent="flex-start"
      scale="s"
    >
      <IconButton
        icon={<BinFilledIcon />}
        isDisabled={false}
        isToggleButton={false}
        label="formatted message"
        onClick={[Function]}
        shape="round"
        size="big"
        theme="default"
        type="button"
      />
      <SecondaryButton
        isToggleButton={false}
        label="formatted message"
        onClick={[MockFunction]}
        theme="default"
        type="button"
      />
      <PrimaryButton
        isDisabled={true}
        isToggleButton={false}
        label="formatted message"
        onClick={[MockFunction]}
        size="big"
        tone="primary"
        type="button"
      />
    </Inline>
  }
  isOpen={true}
  labelPrimaryButton={
    Object {
      "defaultMessage": "Confirm",
      "id": "AppKit.Shared.confirm",
    }
  }
  labelSecondaryButton={
    Object {
      "defaultMessage": "Cancel",
      "id": "AppKit.Shared.cancel",
    }
  }
  onClose={[MockFunction]}
  title="formatted message"
>
  <Horizontal>
    <Stack
      alignItems="stretch"
      scale="s"
    >
      <TextField
        horizontalConstraint="scale"
        isDisabled={false}
        name="name"
        onChange={[MockFunction]}
        title="formatted message"
        value=""
      />
      <MultilineTextField
        defaultExpandMultilineText={true}
        horizontalConstraint="scale"
        isDisabled={false}
        isRequired={true}
        name="value"
        onChange={[MockFunction]}
        placeholder="formatted message"
        renderError={[Function]}
        title="formatted message"
        value=""
      />
    </Stack>
  </Horizontal>
  <br />
  <PredicateDescriptionModal />
</FormModalPage>
`;
